### PR TITLE
Add more iptables rules

### PR DESCRIPTION
**Pull request by baude**
Add rules to allow incoming and outgoing traffic as well as a network
firewall rule.

Signed-off-by: Brent Baude <bbaude@redhat.com>

### CODE REVIEW

File: src/firewall/iptables.rs [hunk 1]
```diff
@@ -22,60 +24,101 @@ pub fn new() -> Result<Box<dyn firewall::FirewallDriver>, Box<dyn Error>> {
...
+                // chains in a table and iterate.  Same is being done in golang implementations
+                let filter_chains = self.conn.list_chains(FILTER)?;
+                if !filter_chains.iter().any(|i| i == PRIV_CHAIN_NAME){
+                    self.conn.new_chain(FILTER, PRIV_CHAIN_NAME).map(|_| debug_chain_create(FILTER, PRIV_CHAIN_NAME))?;
+                } else {
+                    debug_chain_exists(FILTER, PRIV_CHAIN_NAME);
+                }
+
+                //  Create netavark firewall rule
+                //  this is not working ... and must be figured out
```
[the remark this answers is not in the document]
Comment (baude) [reply src/firewall/iptables.rs:81]: wtf, this is a leftover, i rm'd it but maybe rebasing brought it back

File: src/network/types.rs [hunk 2]
```diff
@@ -73,7 +74,7 @@ pub struct PerNetworkOptions {
     pub interface_name: String,
 
     #[serde(rename = "static_ips")]
-    pub static_ips: Option<Vec<String>>,
+    pub static_ips: Option<Vec<IpAddr>>,
```
Comment (flouthoc) [inline src/network/types.rs:77]: thanks for switching this. I wanted to switch but was not sure if some other part of code is using this as string.

File: src/firewall/iptables.rs [hunk 3]
```diff
@@ -22,60 +24,100 @@ pub fn new() -> Result<Box<dyn firewall::FirewallDriver>, Box<dyn Error>> {
...
                         // nat rule does not exists
-                        self.conn.append(NAT, &network_name, &nat_rule).map(|_| {
-                            debug!(
-                                "created iptables nat rule for {}:{}",
-                                &network_name, nat_rule
-                            )
-                        })?;
+                        self.conn
+                            .append(NAT, &network_name, &nat_rule)
+                            .map(|_| debug_rule_create(NAT, &network_name, nat_rule))?;
```
[the remark this answers is not in the document]
Comment (flouthoc) [reply src/firewall/iptables.rs:50]: We can have this in seperate line I think and remove `.map`

[the remark this answers is not in the document]
Comment (baude) [reply src/firewall/iptables.rs:50]: are you maybe misreading the call and its returns?

[the remark this answers is not in the document]
Comment (flouthoc) [reply src/firewall/iptables.rs:50]: @baude my bad i think this is very tiny nit. We shouldn't get blocked on this and revisit this later if we need to.

I think `.map` iterates over all the keys on iter returned by `append()` and we are printing exactly same `debug` for all values in iter. https://doc.rust-lang.org/stable/std/iter/struct.Map.html

Marking this is resolved.

**Pull request merged**